Sources/Shared/DeviceOrientation.swift

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+
+public final class DeviceOrientation {
+    
+  private var windowScene: UIWindowScene? {
+    return UIApplication.shared.connectedScenes.first as? UIWindowScene
+  }
+  
+  public static let shared: DeviceOrientation = DeviceOrientation()
+  
+  public func changeOrientation(_ orientationMask: UIInterfaceOrientationMask) {
+    if #available(iOS 16.0, *) {
+      windowScene?.requestGeometryUpdate(.iOS(interfaceOrientations: orientationMask))
+    } else {
+      var orientation: UIInterfaceOrientation?
+      
+      switch orientationMask {
+      case .portrait:
+        orientation = UIInterfaceOrientation.portrait
+      case .portraitUpsideDown:
+        orientation = UIInterfaceOrientation.portraitUpsideDown
+      case .landscapeRight:
+        orientation = UIInterfaceOrientation.landscapeRight
+      case .landscapeLeft:
+        orientation = UIInterfaceOrientation.landscapeLeft
+      default:
+        orientation = UIInterfaceOrientation.unknown
+      }
+      
+      if let orientation = orientation {
+        UIDevice.current.setValue(orientation.rawValue, forKey: "orientation")
+      }
+    }
+  }
+    
+  public var isLandscape: Bool {
+    if #available(iOS 16.0, *) {
+      return windowScene?.interfaceOrientation.isLandscape ?? false
+    }
+    
+    return UIDevice.current.orientation.isLandscape
+  }
+    
+  public var isPortrait: Bool {
```

Comment:
These don't seem to be used anywhere and will technically be invalid on iPad so probably shouldn't be public

Reply:
Access modifiers are changed at https://github.com/brave/brave-ios/pull/8485/commits/8ac94bc1e30e13dfb6d42a77619106ce97830561